cmd/alertmanager/main.go

```diff
@@ -476,3 +478,26 @@ func md5HashAsMetricValue(data []byte) float64 {
 	copy(bytes, smallSum)
 	return float64(binary.LittleEndian.Uint64(bytes))
 }
+
+func setAlertStatus(inhibitor *inhibit.Inhibitor, marker types.Marker, silences *silence.Silences) func(model.LabelSet) error {
+	return func(labels model.LabelSet) error {
+		inhibitor.Mutes(labels)
+		// TODO(beorn7): The following code is almost exactly replicated in notify/notify.go.
+		sils, err := silences.Query(
+			silence.QState(types.SilenceStateActive),
+			silence.QMatches(labels),
+		)
+		if err != nil {
+			return fmt.Errorf("failed to query silences: %v", err)
+		}
+
```

Comment:
Similar to the code in `notify.go`, don't we have to unset the silence marker of no silences match the alert? I.e.
`n.marker.SetSilenced(labels.Fingerprint())`. (And if that's true, it would prove the point that we should not replicate the code but have it shared somewhere.)

Reply:
This is true, good catch. Updating the code.

Reply:
Thanks. (Would prefer to actually de-replicate the code, but we can do that later.)